pkg/driver/node.go

```diff
@@ -188,9 +189,32 @@ func (ns *node) NodeStageVolume(
 		return nil, status.Error(codes.FailedPrecondition, err.Error())
 	}
 
+	// check the owner node status
+	// if the previous node is not ready, allow
+	// mount to another another node
+	var nodeOk bool
+	if nodeName, ok := instance.Labels["nodeID"]; ok {
+		node, err := ns.client.GetNode(nodeName)
+		if err != nil && !errors.IsNotFound(err) {
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+		if err == nil {
+			nodeOk = isNodeReady(node)
+		}
+	}
+
+	// if the jivaVolume CR has any existing targetPath and the
+	// pevious node is ready abort the stage request
+	if instance.Spec.MountInfo.TargetPath != "" && nodeOk {
+		err = fmt.Errorf("Volume {%v} is already mounted at more than one place: {%v}",
+			reqParam.volumeID, instance.Spec.MountInfo)
+		return nil, status.Error(codes.FailedPrecondition, err.Error())
+
+	}
```

Comment:
This is a good check, in cases where one pod is already scheduled & consuming a volume, and if K8s schedules another pod later to consume the same volume. But this check will not stop in the following case:
- Kubernetes scheduler spins up two pods at the same time on a different nodes to consume the same volume:
  - Now, Kubelet1 on node1 sends `NodeStageVolume` request to driver1 and at same time(no difference in timestamp) Kubelet2 on node2 also sends `NodeStageVolume` request to driver2... Then this check will not help in stop the process... IMO we also need to some checks in [gotgt](https://github.com/gostor/gotgt)...
 

- This can be a case where the user launched deployment with 2 replicas.
- This can be a case where the user launched 2 different deployments at same time.

Reply:
I see that it is handled via 2 stage update during NodeStage request time